Sources/LanguageServerProtocol/SupportTypes/WorkspaceSettings.swift

```diff
@@ -75,3 +80,63 @@ public struct ClangCompileCommand: Codable, Hashable {
     self.workingDirectory = workingDirectory
   }
 }
+
+public struct SourceKitLSPWorkspaceSettings: Decodable, Hashable {
+  private enum CodingKeys: String, CodingKey {
+    case sourcekit_lsp = "sourcekit-lsp"
+    case indexVisibility
+  }
+  
+  // Index visibility setting
+  public var indexVisibility: IndexVisibility? = nil
+  
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    guard let langExtensionDict = try? container.nestedContainer(keyedBy: CodingKeys.self, forKey: .sourcekit_lsp) else {
+        let context = DecodingError.Context(codingPath: decoder.codingPath, debugDescription: "Expected sourcekit-lsp section from configuration change notification")
+        throw DecodingError.dataCorrupted(context)
+    }
+    self.indexVisibility = try? langExtensionDict.decode(IndexVisibility.self, forKey: .indexVisibility)
+  }
+  
+  public init(indexVisibility: IndexVisibility?) {
+    self.indexVisibility = indexVisibility
+  }
+}
+
+/// Index visibility setting, represented by a list of target identifiers and whether the index of transparent target dependencies should be visibile
+public struct IndexVisibility: Decodable, Hashable {
+  
+  private enum CodingKeys: String, CodingKey {
+    case targets
+    case fanout
+  }
+
+  /// all targets included in the scheme
+  public var targets: [BuildTargetIdentifier]
+  public var fanout: Bool
+
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    guard let targetStrings = try? container.decode([String].self, forKey: .targets),
```

Comment:
Instead of complicating the decoding, I think the client (e.g. vscode extension) should provide the `BuildTargetIdentifier` in the correct form.  Since it's just wrapping them in a URI anyway, this should be something the client can do.

With that, this whole struct should be able to to use the automatically generated `Codable` implementation.

Reply:
Hmm there are actually two things here: one is wrapping in the structure with “uri” key, which seems like something the client (eg vscode extension) should do for us.  But there is also the question of target identifier vs name. The identifier is a URI, so should we update the documentation of the config in vscode?

Reply:
I'm gonna expose object directly through extension configuration without any processing in extension for now (so the format has to be correct, something like: `{"uri": "target:\\/\\/path\\/to:target"}`)
if we want to do something fancy I can do it in a separate PR, don't actually know if we can intercept config change, morph the content and relay the config, might need some workaround there.